elide-core/src/main/java/com/yahoo/elide/audit/LogMessage.java

```diff
@@ -1,202 +1,54 @@
 /*
- * Copyright 2015, Yahoo Inc.
+ * Copyright 2019, Yahoo Inc.
  * Licensed under the Apache License, Version 2.0
  * See LICENSE file in project root for terms.
  */
+
 package com.yahoo.elide.audit;
 
-import com.yahoo.elide.annotation.Audit;
-import com.yahoo.elide.core.PersistentResource;
-import com.yahoo.elide.core.RequestScope;
-import com.yahoo.elide.core.ResourceLineage;
 import com.yahoo.elide.security.ChangeSpec;
-
+import com.yahoo.elide.security.PersistentResource;
 import com.yahoo.elide.security.User;
-import de.odysseus.el.ExpressionFactoryImpl;
-import de.odysseus.el.util.SimpleContext;
 
-import java.text.MessageFormat;
-import java.util.List;
 import java.util.Optional;
-import java.util.stream.Collectors;
-
-import javax.el.ELException;
-import javax.el.ExpressionFactory;
-import javax.el.PropertyNotFoundException;
-import javax.el.ValueExpression;
 
 /**
- * An audit log message that can be logged to a logger.
+ * Elide audit entity for a CRUD action.
  */
-public class LogMessage {
-    //Supposedly this is thread safe.
-    private static final ExpressionFactory EXPRESSION_FACTORY = new ExpressionFactoryImpl();
-    private static final String[] EMPTY_STRING_ARRAY = new String[0];
-
-    private final String template;
-    private final PersistentResource record;
-    private final String[] expressions;
-    private final int operationCode;
-    private final Optional<ChangeSpec> changeSpec;
+public interface LogMessage {
 
     /**
-     * Construct a log message that does not involve any templating.
-     * @param template - The unsubstituted text that will be logged.
-     * @param code - The operation code of the auditable action.
+     * Gets message.
+     *
+     * @return the message
      */
-    public LogMessage(String template, int code) {
-        this(template, null, EMPTY_STRING_ARRAY, code, Optional.empty());
-    }
+    public String getMessage();
 
     /**
-     * Construct a log message from an Audit annotation and the record that was updated in some way.
-     * @param audit - The annotation containing the type of operation (UPDATE, DELETE, CREATE)
-     * @param record - The modified record
-     * @param changeSpec - Change spec of modified elements (if logging object change). empty otherwise
-     * @throws InvalidSyntaxException if the Audit annotation has invalid syntax.
+     * Gets operation code.
+     *
+     * @return the operation code
      */
-    public LogMessage(Audit audit, PersistentResource record, Optional<ChangeSpec> changeSpec)
-            throws InvalidSyntaxException {
-        this(audit.logStatement(), record, audit.logExpressions(), audit.operation(), changeSpec);
-    }
+    public int getOperationCode();
```

Comment:
What is an operation code in this context?

Reply:
The original audit design allowed for all log messages to have an audit code.  Most audit systems have something like this (in addition to a log message).

Reply:
This is a generic value correct? Can we mention that in the javadocs?

Reply:
👍